### PR TITLE
Add switch to connected account alert

### PR DESCRIPTION
**Pull request by Gudahtt**
Add alert suggesting that the user switch to a connected account. This alert is displayed when the popup is opened over an active tab that is connected to some account, but not the current selected account. The user can choose to switch to a connected account, or dismiss the alert.

This alert is only shown once per account switch. So if the user repeatedly opens the popup on a dapp without switching accounts, it'll only be shown the first time.

A checkbox allowing the user to dismiss the alert permanently will be added in a subsequent PR.

### CODE REVIEW

**Comment by Gudahtt** [comment]
This depends upon #8524

**Comment by Gudahtt** [comment]
I need to check on a couple of more things before this is ready to review.

* The check for whether the alert was shown already isn't per-dapp right now, but it should be.
* There's a scrollbar that keeps showing up on the home screen that I want to get rid of.

**Comment by Gudahtt** [comment]
<details>
<summary>Screenshots:</summary>
Disregard the scrollbar in these two screenshots - that was removed in a separate PR (#8547)

Multiple accounts:

switch-to-alert

One account:

switch-to-alert-1-account


</details>

File: app/scripts/controllers/app-state.js [hunk 1]
```diff
@@ -107,6 +121,16 @@ export default class AppStateController extends EventEmitter {
     this._resetTimer()
   }
 
+  /**
+   * Sets the "switch to connected" alert as shown for the given origin
+   * @param {string} origin - The origin the alert has been shown for
+   */
+  setSwitchToConnectedAlertShown (origin) {
```
Comment (Gudahtt) [inline app/scripts/controllers/app-state.js:128]: After discussing this with @rachelcope , we've decided to set this alert to "shown" after the user sees the _other_ alert too (the "Unconnected account" alert seen after switching accounts in the popup). This is to prevent them from seeing two similar alerts in a row.

An example of this case is:

* The user opens the popup on a dapp, with a connected account selected
* The user switches to an unconnected account, sees the "unconnected account" alert, then dismisses it
* The user closes the popup
* The user opens the popup again

In that case, currently we'd show the "switch to account" alert. This is what we'd like to change, so that this alert is skipped in this specific scenario.

Comment (Gudahtt) [reply app/scripts/controllers/app-state.js:128]: This has been done in 6b320b70505425a342bb9268f1f180e2da79464e

File: app/scripts/controllers/app-state.js [hunk 2]
```diff
@@ -15,28 +15,42 @@ export default class AppStateController extends EventEmitter {
...
-    }, initState))
+    this.store = new ObservableStore(
+      Object.assign(
+        {
+          connectedStatusPopoverHasBeenShown: true,
+          timeoutMinutes: 0,
+        },
+        initState,
+        {
+          switchToConnectedAlertShown: {},
```
Comment (Gudahtt) [inline app/scripts/controllers/app-state.js:31]: I think this PR is fine to merge as-is, but it might be better to move this stuff into the new `AlertController` created in #8550. If so, maybe it'd be better to get #8550 merged first. Either way I move this afterwards in a separate PR though.

**Review by rachelcope — DISMISSED** [review]
One small design feedback - looks like there is too much left/right padding on the input (making it inconsistently aligned with the title and buttons). 
Otherwise, LGTM!

**Comment by rachelcope** [comment]
Should this have the "don't show this again" functionality?

**Comment by Gudahtt** [comment]
>Should this have the "don't show this again" functionality?

Nope! Though I'm considering updating this PR to include it before merging, now that #8550 is pretty close to being merged.

**Comment by Gudahtt** [comment]
> looks like there is too much left/right padding on the input (making it inconsistently aligned with the title and buttons).

Good catch! I've just updating the input padding to match the text.

<details>
<summary>Here's what it looks like now:</summary>

updated-padding

</details>

**Comment by Gudahtt** [comment]
I've also just realized that I handled the single-account case wrong. It's supposed to look like this if there's one account:

<details>
<summary>Screenshot: </summary>

switch-accounts

</details>

**Comment by Gudahtt** [comment]
The "don't show this again" checkbox has been added to this alert, along with the associated alert settings.

<details>
<summary>Screenshots:</summary>

Alert with checkbox:

with-checkbox

Updated alert settings:

alert-settings

</details>

File: app/scripts/controllers/app-state.js (hunk 2 above)
Comment (Gudahtt) [reply app/scripts/controllers/app-state.js]: This was done in 55f4b3e22e2d2fd63d92dcce09b5b570d875e9fd

**Comment by Gudahtt** [comment]
>I've also just realized that I handled the single-account case wrong.

After discussing this with @rachelcope , the copy for both the single and multiple-account cases has been updated. The alert has also been updated to omit the dropdown if there is just a single account.

<details>
<summary>Here are the updated screenshots:</summary>

Single account:

new-one-account

Multiple accounts:

new-multiple-accounts

</details>

File: ui/app/ducks/alerts/unconnected-account.js [hunk 3]
```diff
@@ -94,7 +94,7 @@ export const dismissAndDisableAlert = () => {
   return async (dispatch) => {
     try {
       await dispatch(disableAlertRequested())
-      await dispatch(setAlertEnabledness(name), false)
+      await dispatch(setAlertEnabledness(name, false))
```
Comment (Gudahtt) [inline ui/app/ducks/alerts/unconnected-account.js:97]: I didn't intend to commit this here - this has been fixed in a separate PR (#8561)

Comment (Gudahtt) [reply ui/app/ducks/alerts/unconnected-account.js]: #8561 has been merged, and this PR has been rebased on `develop`, so this should be gone from the diff now.

**Review by whymarrh — APPROVED** [review]
LGTM!
